dali/pipeline/operators/crop/crop.h

```diff
@@ -75,35 +81,90 @@ class Crop : public SliceBase<Backend>, protected CropAttr {
       case DALI_NFCHW:
         std::tie(F, C, H, W) = std::make_tuple(shape[0], shape[1], shape[2], shape[3]);
         break;
+      case DALI_NDHWC:
+        std::tie(D, H, W, C) = std::make_tuple(shape[0], shape[1], shape[2], shape[3]);
```

Comment:
I think all those "tie = make_tuple" serve little purpose here - tie is meant to assign a tuple assembled elsewhere to an ad-hoc tuple of variables.